luarules/gadgets/unit_xp_range_bonus.lua

```diff
@@ -57,8 +63,8 @@ function gadget:UnitDamaged(unitID, unitDefID, unitTeam, damage, paralyzer, weap
 		return
 	end
 
-	if(UnitDefs[attackerDefID].customParams.rangexpscale ~= nil) then
-		updateList[attackerID] = true	--schedule an update of range next frame when exp has been updated
+	if(XPDefs[attackerDefID]) then		
+		updateList[attackerID] = UnitDefs[unitDefID]	--schedule an update of range next frame when exp has been updated
```

Comment:
just pass the `unitDefID` instead of `UnitDefs[unitDefID]` you can get the `unitDef` later with `XPDefs[unitDefID]` only when you need it.

Reply:
Solved.